cluster/service_registry.go

```diff
@@ -45,10 +47,33 @@ func (sr *ServiceRegistry) Register(ctx context.Context, serviceName, nodeName,
 	}
 
 	key := filepath.Join(servicesPrefix, serviceName, nodeName)
-	if _, err = sr.KV.Put(ctx, key, string(val)); err != nil {
+
+	// TODO: Discuss appripriate lease time, I chose 2 seconds using no imperical data
+	resp, err := sr.cli.Grant(ctx, 2)
+	if err != nil {
+		return fmt.Errorf("failed to create lease for service: %w", err)
+	}
+
+	if _, err = sr.KV.Put(ctx, key, string(val), clientv3.WithLease(resp.ID)); err != nil {
 		return fmt.Errorf("failed to register node: %w", err)
 	}
 
+	keepAliveResp, err := sr.cli.KeepAlive(ctx, resp.ID)
+	if err != nil {
+		return fmt.Errorf("failed to keep service registered: %w", err)
+	}
+	go func() {
+		for {
+			kar := <-keepAliveResp
+			if kar == nil {
+				fmt.Printf("service %s failed to refresh \n", serviceName)
+				break
+
+			}
+			fmt.Printf("service %s refreshed and vaild for(ttl): %d \n", serviceName, kar.TTL)
```

Comment:
We haven't been logging anything in our lib. I think if we want to do so we should do it behind a logger with debug log abilities

Reply:
If we don't end up logging then we can remove this goroutine all together

Reply:
Agreed. Not sure what we want but I'm ok either way. Any thoughts @bkzhang @dipshit ?

Reply:
something like https://github.com/uber-go/zap probably

Reply:
i think that should be a different pr if we want it

Reply:
#43 adds the logger